external-crates/move/crates/move-vm-profiler/src/lib.rs

```diff
@@ -184,18 +179,29 @@ impl GasProfiler {
     }
 
     pub fn to_file(&self) {
-        if !*PROFILER_ENABLED || !self.is_metered() {
+        let config;
+        if let Some(c) = &self.config {
+            config = c
+        } else {
+            return;
+        }
+        if !self.is_metered() {
             return;
         }
-        // Get the unix timestamp
-        let now = std::time::SystemTime::now()
-            .duration_since(std::time::UNIX_EPOCH)
-            .expect("Clock may have gone backwards")
-            .as_nanos();
-
-        let mut p = self.config.base_path.clone();
-        p.push(format!("gas_profile_{}_{}.json", self.profile_name(), now));
+
+        let mut p = (*DEFAULT_PROFILE_OUTPUT_PATH.clone()).to_path_buf();
+        if let Some(f) = &config.full_path {
+            p = f.clone();
+        } else {
```

Comment:
I would recommend pushing the "default full path" logic higher up -- so at that point where you are creating a `VMProfilerConfig`, if you aren't given an output path, you create it there, and allow the profiler to always expect the full path to exist.